src/Tempest/View/src/Renderers/TempestViewCompiler.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\View\Renderers;
+
+use DOMNodeList;
+use Exception;
+use Masterminds\HTML5;
+use Tempest\Core\Kernel;
+use function Tempest\path;
+use Tempest\View\Attributes\AttributeFactory;
+use Tempest\View\Element;
+use Tempest\View\Elements\ElementFactory;
+
+final readonly class TempestViewCompiler
```

Comment:
The idea of the compiler is that it transforms a template into runnable PHP code, before any data processing is done. This makes the output of the compiler fully cacheable.